src/proxy/src/main/java/ee/ria/xroad/proxy/util/VersionUtil.java

```diff
@@ -0,0 +1,56 @@
+package ee.ria.xroad.proxy.util;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.IOUtils;
+
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * Utils for querying version information
+ */
+@Slf4j
+public final class VersionUtil {
+
+    private VersionUtil() { }
+
+    /**
+     * Read installed proxy version information from package
+     * @return version string of format
+     */
+    public static String readProxyVersion() {
+        String version;
+        try {
+            String cmd;
+            if (Files.exists(Paths.get("/etc/redhat-release"))) {
+                cmd = "rpm -q --queryformat '%{VERSION}-%{RELEASE}' xroad-proxy";
+            } else {
+                cmd = "dpkg-query -f '${Version}' -W xroad-proxy";
+            }
+            Process p = Runtime.getRuntime().exec(cmd);
+            p.waitFor();
```

Comment:
This can block forever (e.g. if the process produces more output than fits into the output buffer).